sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3279,11 +3283,33 @@ pi_result cuda_piEventRelease(pi_event event) {
 
 /// Enqueues a wait on the given CUstream for all events.
 /// See \ref enqueueEventWait
+/// TODO: Add support for multiple streams once the Event class is properly
+/// refactored.
 ///
 pi_result cuda_piEnqueueEventsWait(pi_queue command_queue,
                                    pi_uint32 num_events_in_wait_list,
                                    const pi_event *event_wait_list,
                                    pi_event *event) {
+  return cuda_piEnqueueEventsWaitWithBarrier(
```

Comment:
You've implemented the "marker" wait the same as a "barrier" wait giving up on some OOO opportunities. Is this intentional? Maybe mention it in a comment?

FWIW, here is a good summary of the difference between the two:
https://community.intel.com/t5/OpenCL-for-CPU/any-difference-between-clEnqueueBarrierWithWaitList-and/td-p/1158273#.YMjNzV6Dqds.mailto

Reply:
@smaslov-intel That is intentional. The current PI CUDA implementation uses only one stream per PI Queue. CUDA streams are in-order, so there is no difference.
A more optimal solution goes through refactoring the event class, to cover for events associated with multiple streams, and also associate each PI Queue with multiple Streams, but that requires more work and it is outside the scope of this current patch.

Reply:
Thanks for confirming. It'd be good to have a source code comment why the two are the same. Anyways, I already approved as is.